test/games/am2r/gui/test_am2r_cosmetic_patches_dialog.py

```diff
@@ -1,12 +1,53 @@
 from __future__ import annotations
 
+from typing import TYPE_CHECKING
+
+import pytest
 from PySide6 import QtCore
 
 from randovania.games.am2r.gui.dialog.cosmetic_patches_dialog import (
     AM2RCosmeticPatchesDialog,
     hue_rotate_color,
 )
-from randovania.games.am2r.layout.am2r_cosmetic_patches import AM2RCosmeticPatches
+from randovania.games.am2r.layout.am2r_cosmetic_patches import AM2RCosmeticPatches, MusicMode
+
+if TYPE_CHECKING:
+    from pytestqt.qtbot import QtBot
+
+    pass
+
+
+@pytest.mark.parametrize(
+    ("music_start_value", "option_to_click", "music_end_value"),
+    [
+        (MusicMode.VANILLA, "vanilla_music_option", MusicMode.VANILLA),
+        (MusicMode.VANILLA, "type_music_option", MusicMode.TYPE),
+        (MusicMode.VANILLA, "full_music_option", MusicMode.FULL),
+        (MusicMode.TYPE, "vanilla_music_option", MusicMode.VANILLA),
+        (MusicMode.TYPE, "type_music_option", MusicMode.TYPE),
+        (MusicMode.TYPE, "full_music_option", MusicMode.FULL),
+        (MusicMode.FULL, "vanilla_music_option", MusicMode.VANILLA),
+        (MusicMode.FULL, "type_music_option", MusicMode.TYPE),
+        (MusicMode.FULL, "full_music_option", MusicMode.FULL),
+    ],
+)
+def test_change_music_option(
+    skip_qtbot: QtBot, music_start_value: MusicMode, option_to_click: str, music_end_value: MusicMode
+) -> None:
+    cosmetic_patches = AM2RCosmeticPatches(music=music_start_value)
+
+    dialog = AM2RCosmeticPatchesDialog(None, cosmetic_patches)
+    skip_qtbot.addWidget(dialog)
+    str_to_option_map = {
+        "vanilla_music_option": dialog.vanilla_music_option,
+        "type_music_option": dialog.type_music_option,
+        "full_music_option": dialog.full_music_option,
+    }
+    radio_button = str_to_option_map[option_to_click]
+
+    skip_qtbot.mouseClick(radio_button, QtCore.Qt.LeftButton)
```

Comment:
I like mypy :)
Is that really a thing or is it only defined in `QtCore.Qt.MouseButton.LeftButton` ?
(I wonder why the tests are not failing if that is true)

Reply:
PySide at some point cleaned up and exposed enums correctly, but it was previously C-like where the values weren't scoped to the enum type. The type hints are only present with the "good" name.

Reply:
Maybe i should start reading this properly, haha. For some reason I thought all this time it was complaining about `skip_qtbot.mouseClick` and just completely ignored it.

I cannot find `qt.leftbutton` anywhere, so i'm assuming that mypy is correct. 


EDIT: somethign that dark seems to confirm